src/ert/gui/model/snapshot.py

```diff
@@ -273,7 +285,13 @@ def _add_partial_snapshot(self, partial: PartialSnapshot, iter_: int) -> None:
 
             return
 
-    def _add_snapshot(self, snapshot: Snapshot, iter_: int) -> None:
+    def _process_full_snapshot(self, snapshot: Snapshot, iter_: int):
+        """Add a new snapshot for an iteration, or completely overwrite the current one.
```

Comment:
"or" here makes it look conditional whether it is overwritten. You probably mean it will overwrite if there is something already for a particular iteration. You can maybe say "Set snapshot for a given iteration" and comment (in paranthesis?) that anything existing will be overwritten.